src/chainparams.cpp

```diff
@@ -72,6 +72,7 @@ class CMainParams : public CChainParams {
         consensus.BIP66Height = 363725; // 00000000000000000379eaa19dce8c9b722d46ae6a57c2f1a988119488b50931
         consensus.CSVHeight = 419328; // 000000000000000004a1b34462cb8aeebd5799177f7a29cf28f2d1961716b5b5
         consensus.SegwitHeight = 481824; // 0000000000000000001c8018d9cb3b742ef25114f27563e3fc4a1902167f9893
+        consensus.TaprootHeight = 709632; // 0000000000000000000687bca986194dc2c1f949318629b44bb54ec0a94d8244
```

Comment:
710adad788898b6e0f31816017165af4fdbc2aad: Why do we care about main/testnet params if we are disabling mainnnet?

Reply:
Many of the unit tests use the mainnet params so getting rid of the main params entirely doesn't work well, and if the params are going to be there at all, they might as well at least be correct. This patch is cherry-picked from bitcoin/bitcoin#23505.